### DISCUSSION TITLE
Storing data per user/session

### DISCUSSION THREAD

**Discussion by ulrichbi** [zauberzeug/nicegui#449] [Q&A] [answered]
Hi,

maybe this is a stupid question, but I am new to NiceGUI:

If I store data in global variables then does this also work if several users are accessing the pages? So would the content in "content" be separated by user/session?

```py
from nicegui import app, ui

content = {}

@ui.page('/two')
def page_two():
    ui.label("Yes" if content["s"].value else "No")
    ui.link('back', '/')

@ui.page('/')
def main_page():
    content["s"] = ui.switch('switch')
    ui.link('next ', page_two)

ui.run()
```

**Comment by falkoschindler** [comment, marked as answer]
Good question, @ulrichbi!
No, the value of global variables like `content` is shared across all users and sessions.
[Here](https://nicegui.io/reference#sessions) we show an example of how to access a `session` dictionary, which is a separate storage per session. But we noticed that this is not very convenient and we are currently thinking about a better approach. See #373 for a similar discussion.
